tsdb/tsm1/reader_range_iterator.go

```diff
@@ -105,9 +78,58 @@ func (b *TimeRangeIterator) HasData() bool {
 	return false
 }
 
+type timeRangeBlockReader struct {
+	r     *TSMReader
+	iter  *TSMIndexIterator
+	tr    TimeRange
+	err   error
+	stats cursors.CursorStats
+
+	// temporary storage
+	trbuf []TimeRange
+	buf   []byte
+	a     cursors.TimestampArray
+}
+
+func (b *timeRangeBlockReader) Err() error {
+	if b.err != nil {
+		return b.err
+	}
+	return b.iter.Err()
+}
+
+// Key reports the current key.
+func (b *timeRangeBlockReader) Key() []byte {
+	return b.iter.Key()
+}
+
+// Type reports the current block type.
+func (b *timeRangeBlockReader) Type() byte {
+	return b.iter.Type()
+}
+
+func (b *timeRangeBlockReader) getEntriesAndTombstones() ([]IndexEntry, []TimeRange) {
+	if b.err != nil {
+		return nil, nil
+	}
+
+	e := excludeEntries(b.iter.Entries(), b.tr)
```

Comment:
I wouldn't mind understand this a bit more. Doesn't `excludeEntries` only return entries that are _completely_ overlapped by `b.tr`? Wouldn't you be interested in entries that are _partially_ overlapped by `b.tr`?

Reply:
Your assumption is correct, and `excludeEntries` does handle partially covered, per these two tests.

https://github.com/influxdata/influxdb/blob/0ae8bebd755cf4aa579b9c38e6ea4792dfdbe53a/tsdb/tsm1/reader_range_iterator_test.go#L337-L352

Which is effectively:

```
+------------------------------------------+
|                 Entries                  |
+----------+---+----------+-----+----------+
| [0, 10]  |   | [12, 15] |     | [19, 21] |
+----------+   +----------+     +----------+
             +----------+                   
             |Time Range|                   
             +----------+                   
             | [11, 13] |                   
             +----------+                   
                   |                        
              excludeEntries                
                     |                      
                     v                      
               +----------+                 
               | [12, 15] |                 
               +----------+                 
                                            
                                            
+------------------------------------------+
|                 Entries                  |
+----------+---+----------+-----+----------+
| [0, 10]  |   | [12, 15] |     | [19, 21] |
+----------+   +----------+     +----------+
           +------------+                   
           | Time Range |                   
           +------------+                   
           |  [10, 13]  |                   
           +------------+                   
                  |                         
                  |                         
        excludeEntries                      
             |                              
             v                              
+-------------------------+                 
|         Entries         |                 
+----------+---+----------+                 
| [0, 10]  |   | [12, 15] |                 
+----------+   +----------+                              
```